src/SDL_utils.c

```diff
@@ -121,7 +121,40 @@ bool SDL_endswith(const char *string, const char *suffix)
     return false;
 }
 
-// Assume we can wrap SDL_AtomicInt values and cast to Uint32
+bool SDL_ShouldInit(SDL_InitState *state)
+{
+    while (SDL_GetAtomicInt(&state->status) != SDL_INIT_STATUS_INITIALIZED) {
+        if (SDL_CompareAndSwapAtomicInt(&state->status, SDL_INIT_STATUS_UNINITIALIZED, SDL_INIT_STATUS_INITIALIZING)) {
+            state->thread = SDL_GetCurrentThreadID();
+            return true;
+        }
+
+        // Wait for the other thread to complete transition
+        SDL_Delay(1);
```

Comment:
I'm not a huge fan of delaying in a loop, vs the AtomicMutex thing I posted yesterday (even if it was just one global lock that all ShouldInits reuse), but honestly actually hitting this call is extremely unlikely, ever, so it's probably fine.

Reply:
Yeah, I thought about creating a condition variable and broadcasting when init was complete, which had its own problems with initialization, and massively increased complexity, so this seemed like the safest and easiest solution.